libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -34,8 +34,10 @@ namespace solidity
 
 NameAndTypeResolver::NameAndTypeResolver(
 	vector<Declaration const*> const& _globals,
+	map<ASTNode const*, shared_ptr<DeclarationContainer>>& _scopes,
```

Comment:
It has to be longer that the lifetime of this object, of course the shared pointers live on longer.